Fix links

lomamech opened the pull request:
fix(link) …
- Fix the link betwen Debtor-group and Patient registry 
- Fix the link betwen Patient-group and Patient registry
 - Update test/data.sql remove Employee/Test Debtor because this debtor   isn't a Patient

closes #1873

lomamech committed:
fix(link)
- Fix the link betwen Debtor-group and Patient registry
- Fix the link betwen Patient-group and Patient registry
- Update test/data.sql remove Employee/Test Debtor because this debtor
  isn't a Patient

lomamech committed:
Resolve conflict

lomamech committed:
test(check link)
- Remove dysfonctionnement test E2E

jniles commented on test/end-to-end/patient/invoice/invoice.page.js:
```diff
@@ -30,8 +30,8 @@ function PatientInvoicePage() {
     findPatient.findById('PA.TPA.1');
 
-    // set the date to early this year
-    dateEditor.set(new Date('2016-02-02'));
```
Why did this change need to take place?  Also, the date no longer matches the comment :(

lomamech replied:
It is just so that verification can not be done with empty registers

jniles commented on test/end-to-end/patient/registry.search.js:
```diff
@@ -64,7 +63,7 @@ function PatientRegistrySearch() {
 
   it(`should find three patient with Debtor Group "Second Test Debtor Group"`, () => {
-    const NUM_MATCHING = 2;
+    const NUM_MATCHING = 3;
```
Nice :+1:  The comment now makes sense!

jniles commented on test/end-to-end/verificationLinks/verificationLinks.spec.js:
```diff
@@ -0,0 +1,62 @@
+    .all(by.repeater('(rowRenderIndex, row) in rowContainer.renderedRows track by $index'));  
+
+  it('Check link betwen Patient Groups -> Patient Registry', () => {
+    browser.get('#!/patients/groups');
```
I think it would be better to use `helpers.navigate()` since it using hash-based routing.  `browser.get()` may slow down loading the whole page by reloading angular.

The [documenation](http://www.protractortest.org/#/api?view=ProtractorBrowser.prototype.get) is not clear on this though.

jniles commented on test/data.sql:
```diff
@@ -249,19 +249,19 @@ INSERT INTO `patient_group` VALUES
-  (HUID('be0096dd-2929-41d2-912e-fb2259356fb5'),HUID('4de0fe47-177f-4d30-b95f-cff8166400b4'),'Employee/Test Debtor');
+  (HUID('a11e6b7f-fbbb-432e-ac2a-5312a66dccf4'),HUID('4de0fe47-177f-4d30-b95f-cff8166400b4'),'Patient/1/Patient');
 
 INSERT INTO `patient` VALUES
```
Are you changing the registration date here, or the dob?

lomamech replied:
I had changed the registration date

jniles replied:
Okay, that's fine.

jniles replied:
Okay, that's fine.

jniles commented on client/src/modules/debtors/groups.list.html:
```diff
@@ -58,7 +58,7 @@ <h5 style="margin-top : 0px;"></span><span translate> FORM.LABELS.ACCOUNT_NUMBER
 
                   <div class="col-md-6">
-                    <a ui-sref="patientRegistry({ filters : { debtor_group_uuid : debtorGroup.uuid }})"><span class="fa fa-bars" aria-hidden="true"></span> <b>{{debtorGroup.total_debtors}}</b> <span translate> DEBTOR_GROUP.DEBTOR.SUBSCRIBED </span></a>
+                    <a ui-sref="patientRegistry({ filters : { key : 'debtor_group_uuid', value : debtorGroup.uuid }})"><span class="fa fa-bars" aria-hidden="true"></span> <b>{{debtorGroup.total_debtors}}</b> <span translate> DEBTOR_GROUP.DEBTOR.SUBSCRIBED </span></a>
```
Nice :+1:

lomamech committed:
perf(fix links)
- Update comment in e2e test for Invoice
- Update verificationLinks e2e use helpers.navigate instead browser.get
